CenterForOpenScience/ember-osf-web
# [ENG-3300] Properly encode provider name in GA event

-   Ticket: [ENG-3300](https://openscience.atlassian.net/browse/ENG-3300)
-   Feature flag: n/a

## Purpose

We were sending the string '@provider.name' to Google Analytics instead of the provider's name in a few instances. This corrects that.

## Summary of Changes

1. Use @provider.name in registry discovery results
2. Use @provider.name in sort dropdown
3. Use @provider.name in sharing control

## Screenshot(s)

<img width="474" alt="Screen Shot 2021-11-08 at 12 18 16 PM" src="https://user-images.githubusercontent.com/6599111/140798614-2c63eb5b-e22e-4043-bcb1-f47b5f8730cd.png">
<img width="490" alt="Screen Shot 2021-11-08 at 12 17 42 PM" src="https://user-images.githubusercontent.com/6599111/140798616-34b1a580-3418-4e6a-be45-e3074b98b62a.png">
<img width="455" alt="Screen Shot 2021-11-08 at 12 17 10 PM" src="https://user-images.githubusercontent.com/6599111/140798617-bbc83601-9747-40ac-a2b8-ac37b2fa7b82.png">


## Side Effects

No, this is isolated.

## QA Notes

Should just need to check the `collect` network request to ensure that the provider name is being included in the request for analytics information collected on a branded provider's discover page.
